roles/deploy/handlers/main.yml

```diff
@@ -1,27 +1,63 @@
 ---
 
-- name: restart delayed_job
+- name: check services
+  service_facts:
+  listen: update jobs queue
+
+- name: check sidekiq presence in gemfile
+  lineinfile:
+    dest: "{{ build_path }}/Gemfile"
+    regex: "^gem .*sidekiq"
+    state: absent
+  check_mode: yes # Don't actually modify the file
+  become: yes
+  register: sidekiq_requirement
+  ignore_errors: true
+  listen: update jobs queue
+
+- name: restart sidekiq
   service:
-    name: delayed_job_{{ app }}
+    name: sidekiq
     state: restarted
   become: yes
   become_user: root
+  when:
+   - "'sidekiq.service' in services"
+   - sidekiq_requirement.changed
+  listen: update jobs queue
 
-- name: get crontab entries for unicorn user # noqa 305
-  shell: crontab -l
-  register: crontab_entries
+- name: restart delayed_job unless using sidekiq
+  service:
+    name: delayed_job_{{ app }}
+    state: restarted
   become: yes
-  become_user: "{{ unicorn_user }}"
-  changed_when: false
-  failed_when: crontab_entries.rc > 1
-  listen: update whenever
+  become_user: root
+  when:
+   - "'delayed_job_openfoodnetwork.service' in services and services['delayed_job_openfoodnetwork.service']['status'] == 'enabled'"
+   - not sidekiq_requirement.changed
+  listen: update jobs queue
+
+- name: pause to allow final jobs to complete
```

Comment:
so stopping the service in the task below doesn't wait for the action to finish? manually I think it does.

Reply:
The first time we switch from DelayedJob to Sidekiq it should wait 30 seconds before stopping and disabling DelayedJob. I'll give it another test...

Reply:
I missed a bit here. I just updated and re-tested and it's working correctly with the 30-seond wait :+1:

It should only do this once. After delayed job is disabled it'll ignore those steps.

Reply:
Triple-checked :heavy_check_mark:

Reply:
![Screenshot from 2021-05-31 12-04-13](https://user-images.githubusercontent.com/9029026/120184107-e4462680-c210-11eb-99cb-b1cd06ac810e.png)

Reply:
That's some silky-smooth ops right there :sunglasses: